04_numeronym/alanhillgemann/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+func numeronyms(vals ...string) []string {
+	numeronymStrings := make([]string, 0)
+	for _, val := range vals {
+		if val == "" {
+			numeronymStrings = append(numeronymStrings, "")
+		} else {
+			runes := []rune(val)
+			firstChar := string(runes[0])
+			middleChar := ""
+			lastChar := ""
+			if len(runes) > 1 {
+				lastChar = string(runes[len(runes)-1])
+			}
+			if len(runes) == 3 {
+				middleChar = string(runes[1])
+			}
+			if len(runes) > 3 {
```

Comment:
If you change this to `>= 3` or `> 2`, it covers the second case (`== 3`).

Reply:
Improved by nesting if statements